pkg/operator/ceph/object/cosi/controller.go

```diff
@@ -118,8 +118,18 @@ func (r *ReconcileCephCOSIDriver) Reconcile(context context.Context, request rec
 func (r *ReconcileCephCOSIDriver) reconcile(request reconcile.Request) (reconcile.Result, cephv1.CephCOSIDriver, error) {
 	cephCOSIDriver := &cephv1.CephCOSIDriver{}
 
+	// Check No of instances of CephCOSIDriver
+	cephCOSIDriverList := &cephv1.CephCOSIDriverList{}
+	err := r.client.List(r.opManagerContext, cephCOSIDriverList)
+	if err != nil {
+		return reconcile.Result{}, *cephCOSIDriver, errors.Wrap(err, "failed to list CephCOSIDriver")
+	}
+	if len(cephCOSIDriverList.Items) > 1 {
+		return reconcile.Result{}, *cephCOSIDriver, errors.New("more than one instance of CephCOSIDriver found")
+	}
+
 	// Fetch the CephCOSIDriver instance
-	err := r.client.Get(r.opManagerContext, request.NamespacedName, cephCOSIDriver)
+	err = r.client.Get(r.opManagerContext, request.NamespacedName, cephCOSIDriver)
```

Comment:
Since you listed the drivers above, just use the item that was queried instead of again querying with the `Get()`
```suggestion
	cephCOSIDriver := cephCOSIDriverList.Items[0]
```